src/formatDate.js

```diff
@@ -9,6 +9,41 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+
+  const splitedDate = date.split(oldSeparator);
+
+  const newDate = [];
+
+  let yearFormat = toFormat.includes('YY') ? 'YY' : 'YYYY';
+
+  for (let i = 0; i < splitedDate.length; i++) {
+    const oldPart = fromFormat[i];
+    let index = toFormat.indexOf(oldPart);
+
+    if(oldPart.includes('Y') && oldPart !== yearFormat) {
+      index = toFormat.indexOf(yearFormat);
+    }
+
+    const newPart = toFormat[index];
+    
+    if (oldPart.length > newPart.length) {
+      splitedDate[i] = splitedDate[i].split('').slice(-2).join('');
```

Comment:
When converting a year from 'YYYY' to 'YY', directly slicing the last two digits is a good approach. However, this assumes the year will always be in the same position, which might not hold true for different formats. Consider finding the index of the year part explicitly.